code/game/machinery/computer/Operating.dm

```diff
@@ -95,7 +94,27 @@
 		usr.set_machine(src)
 	return
 
-
 /obj/machinery/computer/operating/process()
 	if(..())
 		src.updateDialog()
+	update_icon()
+
+/obj/machinery/computer/operating/update_icon()
+	..()
+	icon_state = initial(icon_state)

```

Comment:
1. You can just assign a base_icon_state instead of calling initial, especially if it is triggering once per tick. Initial isn't extremely expensive but it isn't necessary to be wasting cpu on it when you already have the object you can read a variable from.

2. Power_change() calls update icon anyway, so I think it would be preferable to set your else statement to a distinct proc to be triggered in process if (! (stat & (BROKEN | NOPOWER)))

3. It's probably preferable to first check in the else statement that the condition has changed from what it was, before resetting the icon state. I can't say for certain whether or not byond redraws the image simply from setting the icon state equal to something.